spyder/plugins/projects/plugin.py

```diff
@@ -392,10 +392,26 @@ def on_close(self, cancelable=False):
     def unmaximize(self):
         """Unmaximize the currently maximized plugin, if not self."""
         if self.main:
-            if (self.main.last_plugin is not None and
-                    self.main.last_plugin._ismaximized and
-                    self.main.last_plugin is not self):
-                self.main.maximize_dockwidget()
+            layouts = self.get_plugin(Plugins.Layout)
+            last_plugin = layouts.get_last_plugin()
+            is_maximized = False
+
+            if last_plugin is not None:
+                try:
+                    # New API
+                    is_maximized = (
+                        last_plugin.get_widget().get_maximized_state()
+                    )
+                except AttributeError:
+                    # Old API
+                    is_maximized = last_plugin._ismaximized
+
+            if (
+                last_plugin is not None
+                and is_maximized
+                and last_plugin is not self
+            ):
+                layouts.unmaximize_dockwidget()
```

Comment:
Should this logic be in the `Layout` plugin? Maybe a method with a signature like `unmaximize_other_dockwidget(self, plugin_instance)` or maybe could e worthy to modify the current `unmaximize_dockwidget` method to recieve a kwarg, something like `check_plugin_instance=None`. With that here the kwarg could be used to pass `self` and then this will become `layouts.unmaximize_dockwidget(check_plugin_instance=self)`

Reply:
Good idea. I'll go with your first suggestion because I find it easier to understand.